falcon/request.py

```diff
@@ -305,21 +350,30 @@ def __init__(self, env, options=None):
 
         # NOTE(kgriffs): Wrap wsgi.input if needed to make read() more robust,
         # normalizing semantics between, e.g., gunicorn and wsgiref.
-        if _maybe_wrap_wsgi_stream:
-            if isinstance(self.stream, (NativeStream, InputWrapper,)):
-                self._wrap_stream()
-            else:
-                # PERF(kgriffs): If self.stream does not need to be wrapped
-                # this time, it never needs to be wrapped since the server
-                # will continue using the same type for wsgi.input.
-                _maybe_wrap_wsgi_stream = False
+        if not Request._wsgi_input_type_known:
+            Request._always_wrap_wsgi_input = isinstance(
+                env['wsgi.input'],
+                (NativeStream, InputWrapper)
+            )
+
+            Request._wsgi_input_type_known = True
+
+        if Request._always_wrap_wsgi_input:
+            # TODO(kgriffs): In Falcon 2.0, stop wrapping stream since it is
+            # less useful now that we have bounded_stream.
+            self.stream = self._get_wrapped_wsgi_input()
+            self._bounded_stream = self.stream
+        else:
+            self.stream = env['wsgi.input']
+            self._bounded_stream = None  # Lazy wrapping
 
         # PERF(kgriffs): Technically, we should spend a few more
         # cycles and parse the content type for real, but
         # this heuristic will work virtually all the time.
         if (self.options.auto_parse_form_urlencoded and
                 self.content_type is not None and
-                'application/x-www-form-urlencoded' in self.content_type):
+                'application/x-www-form-urlencoded' in self.content_type and
+                self.method == 'POST'):
```

Comment:
ehhh... If you did that, it would also apply for DELETE and HEAD as well, so I think it's fine.